iorgen/parser_julia.py

```diff
@@ -0,0 +1,299 @@
+                name,
+                ", ".join(
+                    "{}::{}".format(var_name(i.name), type_str(i.type, self.input))
+                    for i in self.input.input
+                ),
+            )
+        )
+        if reprint:
+            for var in self.input.input:
+                self.method.extend(self.print_lines(var_name(var.name), var.type, 1))
+        else:
+            self.method.extend(
+                textwrap.wrap(
+                    self.input.output,
+                    79,
+                    initial_indent=INDENTATION + "# " + "TODO ",
+                    subsequent_indent=INDENTATION + "# ",
+                )
+            )
+        self.method.append("end")
+        self.main.append(
+            "{}({})".format(
+                name, ", ".join([var_name(i.name) for i in self.input.input])
+            )
+        )
+
+    def print_lines(self, name: str, type_: Type, indent_lvl: int = 0) -> List[str]:
+        """Print the content of a var that holds in one or more lines"""
+        indent = INDENTATION * indent_lvl
+        if type_.fits_in_one_line(self.input.structs):
+            return [indent + print_line(name, type_, self.input)]
+        if type_.main == TypeEnum.LIST:
+            assert type_.encapsulated is not None
+            inner = "iT" + str(abs(hash(name)))  # unique name
+            return (
+                [indent + "for {} in {}".format(inner, name)]
+                + self.print_lines(inner, type_.encapsulated, indent_lvl + 1)
+                + [indent + "end"]
+            )
+        assert type_.main == TypeEnum.STRUCT
+        lines = []
+        for i in self.input.get_struct(type_.struct_name).fields:
+            lines.extend(
+                self.print_lines(
+                    "{}.{}".format(name, var_name(i.name)), i.type, indent_lvl
+                )
+            )
+        return lines
+
+    def content(self) -> str:
+        """Return the parser content"""
+        output = ""
+        struct_parsers = [self.def_read_struct(struct) for struct in self.input.structs]
+        if self.import_parse:
+            output += "import Base: parse\n\n"
+        for struct in self.input.structs:
+            for line in self.decl_struct(struct):
+                output += line + "\n"
+            output += "\n"
+        for line in self.method:
+            output += line + "\n"
+        if self.method:
+            output += "\n"
+        for parser in struct_parsers:
+            for line in parser:
+                output += line + "\n"
+            output += "\n"
+        for line in self.main:
+            output += line + "\n"
+        return output
+
+
+def gen_julia(input_data: Input, reprint: bool = False) -> str:
+    """Generate a Julia code to parse input"""
+    parser = ParserJulia(input_data)
+    for var in input_data.input:
+        parser.main.extend(parser.read_var(var))
+    parser.call(reprint)
+    return parser.content()
+
+
+RESERVED = [
```

Comment:
You’ll probably need `parse` here as well (or in a second array if you want to keep the keywords seperated)